File: pythonspace/Solution16.py
# Codility - TapeEquilibrium
def solution(A):
    # write your code in Python 3.6
    start = A[0]
    end = sum(A[1:])
    answer = abs(start - end)
    
    for i in range(1, len(A)-1):
        start += A[i]
        end -= A[i]
        answer = min(answer, abs(start - end))

    return answer

print(solution([1,2,3,4,10]))
print(solution([-1,-2]))
print(solution([-10, -20, -30, -40, 100]))

File: javaspace/Solution16.java
import java.util.*;

// Codility - Triangle
public class Solution16 {
    public static void main(String[] args) {
        int[] A = {10,20,22,3,1,4};
        
        System.out.println(solution(A));
    }

    public static int solution(int[] A) {
        // write your code in Java SE 11
        int aLength = A.length;
        if (aLength < 3) {
            return 0;
        }
        
        Arrays.sort(A);

        for (int i=0; i < aLength-2; i++) {
            long P = A[i];
            long Q = A[i+1];
            long R = A[i+2];
            if (P + Q > R) {
                return 1;
            }
        }
        return 0;
    }
}
